_posts/2021-06-29-fed-talk-s01e04-react-components.md

```diff
@@ -0,0 +1,545 @@
+---
+layout: post
+font: serif
+title: "FED Talk! Episode 4: React Components"
+description: "In today’s episode we talk all things React Components, taking you from novice to pro!"
+date: 2021-06-29
+highlight: monokai
+image: /img/fed-talk/s01e04/cover-ep4-740.png
+author: Quintin Maseyk
+tags: [react, react Components, Component composition]
+---
+
+![Front-End Development Talk: Episode 4 - React Components](/img/fed-talk/s01e04/cover-ep4-740.png)
+
+In today's episode we will step through how to implement good React Components, covering:
+
+* Component Fundamentals
+* Component Hooks
+* Component Composition
+
+**Let's get started!**
+
+---
+
+**Table of Contents:**
+
+- [Component Fundamentals](#component-fundamentals)
+- [Component Lifecycle](#component-lifecycle)
+  * [Component Mount Phase](#component-mount-phase)
+  * [Component Un-Mount Phase](#component-un-mount-phase)
+- [Component Hooks](#component-hooks)
+  * [useEffect](#useeffect)
+  * [useCallback](#usecallback)
+  * [useMemo](#usememo)
+    + [Scenario 1: Expensive Computations](#scenario-1-expensive-computations)
+    + [Scenario 2: Expensive Renders](#scenario-2-expensive-renders)
+    + [Scenario 3: Prop Drilling Madness](#scenario-3-prop-drilling-madness)
+- [Component Composition](#component-composition)
+  * [UI Component (low-level element)](#ui-component-low-level-element)
+  * [Parent Child Relationship](#parent-child-relationship)
+  * [Children Prop](#children-prop)
+- [:pray: Closing](#pray-closing)
+
+---
+
+## Component Fundamentals
+
+> :warning: React Components can be a sensitive topic, so if you get offended jump to the next chapter.
+
+Components allow us to abstract logic and state away from the the view (DOM). Its existence enables us to think generically and build abstractions, thus, build repeatable testable bits of code.
```

Comment:
I would change some of the grammar here:

"Their existence enables us to think generically and build abstractions, allowing for easily testable and reusable bits of code."